### PR TITLE
WIP: feat(generate-config): create config with existing service account

### PR DESCRIPTION
**Pull request by rkpattnaik780**
`rhoas generate-config` should not create service-account.

### Verification Steps

1. Generate configuration by specifying `--type` to be `configmap`
```
rhoas generate-config --type=configmap
```
2. It should create a configmap object.
3. Unlike previous version, `rhoas generate-config` shouldn't create a service account.

### ToDo:

- Add validation for flag combinations
- Add validation to check if the service account credentials provided are valid
- Add a message for the breaking change

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation change
- [ ] Other (please specify)

### CODE REVIEW

**Comment by wtrocki** [comment]
Service accounts (generally any credentials cannot be passed thru flags due to bash history. We had initial design for that case in docs. Idea was to use json file provided from service account command

**Comment by rkpattnaik780** [comment]
> We had initial design for that case in docs. Idea was to use json file provided from service account command. WDYT?

I was planning to go with it in the beginning, but it occurred to me that credntial can be in some other format too, for which user wll have to reconstruct it, that could be bit time taking, hence dropped the idea to have it as main method to get the credentials.

> Service accounts (generally any credentials) cannot be passed thru flags due to them being saved in plain form to bash history. 

I will agree, hence there are twoflags - `--client-secret-stdin` that will take up the secret from standard input and `--client-secret` to pass as argument for CI/CD environment. WDYT?

**Comment by wtrocki** [comment]
We can use json file as input and env variables presence(user running .env file) Do we need more options?

**Comment by wtrocki** [comment]
> some other format too

I would not be worried about it because:

- Json is default 
- User will generate service account specifically for this purpose
- We will add validation and description
- guide/docs will reflect it

**Comment by wtrocki** [comment]
Since we designed that command many things changed:

- we now have secret in service accounts as output - enabling openshift integration there


I'm wondering if we actually approached problem right:

Do we really want to supply users service account?
Remember that we would have that already in user desired format. Maybe we just want to stop creating service account as part of generate command?

That will have numerous benefits:
- Generate can be called many times without side effects and paired with service account file
- We can better reflect openshift story:
Credentials would be in secret but urls can be configmap

**Comment by wtrocki** [comment]
@rkpattnaik780 we need to put some research on this as there are different opinions how we should do it.

Mainly there is big trend in current users to disable sa creation in context generate command.
Another option is to provide flag for that use case etc. We might need to drive some proper pros and cons and see how this affects our future roadmap (helm etc.)

**Comment by wtrocki** [comment]
FYI (low priority) @jackdelahunt @dimakis

File: pkg/cmd/generate/templates.go [hunk 1]
```diff
@@ -15,57 +15,39 @@ var (
...
-		{{end}}
-		## Authentication Configuration
-		rhoasClientID={{.ClientID}}
-		rhoasClientSecret={{.ClientSecret}}
-		rhoasOauthTokenUrl={{.TokenURL}}
+		serviceRegistryCompatPath=` + registrycmdutil.REGISTRY_COMPAT_PATH + `{{end}}
 	`)
 
-	templateSecret = heredoc.Doc(`
+	templateConfigMap = heredoc.Doc(`
```
Comment (rkpattnaik780) [inline pkg/cmd/generate/templates.go:39]: @wtrocki 
This PR changes `generate-config` command to skip creating a service account. And replaces secret with configmap.

**Comment by wtrocki** [comment]
Is that good to be verified?

**Comment by rkpattnaik780** [comment]
> Is that good to be verified?

Yes, I will be updating help texts and examples in a while.

File: pkg/cmd/generate/configurations.go [hunk 2]
```diff
@@ -15,16 +15,16 @@ const (
...
 
 var (
-	envConfig            = template.Must(template.New(envFormat).Parse(templateEnv))
-	jsonConfig           = template.Must(template.New(jsonFormat).Parse(templateJSON))
-	propertiesConfig     = template.Must(template.New(propertiesFormat).Parse(templateProperties))
-	secretTemplateConfig = template.Must(template.New(secretFormat).Parse(templateSecret))
+	envConfig             = template.Must(template.New(envFormat).Parse(templateEnv))
+	jsonConfig            = template.Must(template.New(jsonFormat).Parse(templateJSON))
+	propertiesConfig      = template.Must(template.New(propertiesFormat).Parse(templateProperties))
+	congMapTemplateConfig = template.Must(template.New(configmapFormat).Parse(templateConfigMap))
```
Comment (wtrocki) [inline pkg/cmd/generate/configurations.go:27]: image 
;)
```suggestion
	kongMapTemplateConfig = template.Must(template.New(configmapFormat).Parse(templateConfigMap))
```

**Review by wtrocki — APPROVED** [review]

**Comment by wtrocki** [comment]
I think we might need to add some documentation/msg. Something like:

`Generate config finished successfully. You can now create new service accounts in order to connect to the service`

File: pkg/core/localize/locales/en/cmd/generate_config.en.toml [hunk 3]
```diff
@@ -43,4 +43,8 @@ one = 'file {{.FilePath}} already exists. Use --overwrite to overwrite the file,
 one='No services available to generate configurations'
 
 [generate.log.info.credentialsSaved]
-one='Configurations successfully saved to "{{.FilePath}}"'
\ No newline at end of file
+one='''
+Configurations successfully saved to "{{.FilePath}}"
+
+You can now create new service accounts or use existing ones to connect to the service(s)
```
Comment (wtrocki) [inline pkg/core/localize/locales/en/cmd/generate_config.en.toml:49]: As user I'm wondering how.
Can we be more specific.

Run rhoas service-account create etc.

**Review by wtrocki — APPROVED** [review]

**Comment by jackdelahunt** [comment]
Verification steps work 👍

**Pull request merged**